x/wasm/keeper/keeper.go

```diff
@@ -1161,30 +1144,38 @@ func (c BankCoinTransferrer) TransferCoins(parentCtx sdk.Context, fromAddr sdk.A
 	return nil
 }
 
-var _ CoinPruner = CoinBurner{}
+var _ CoinPruner = VestingCoinBurner{}
 
-// CoinBurner default implementation for CoinPruner to burn the coins
-type CoinBurner struct {
+// VestingCoinBurner default implementation for CoinPruner to burn the coins
+type VestingCoinBurner struct {
 	bank types.BankKeeper
 }
 
-// NewCoinBurner constructor
-func NewCoinBurner(bank types.BankKeeper) CoinBurner {
+// NewVestingCoinBurner constructor
+func NewVestingCoinBurner(bank types.BankKeeper) VestingCoinBurner {
 	if bank == nil {
 		panic("bank keeper must not be nil")
 	}
-	return CoinBurner{bank: bank}
+	return VestingCoinBurner{bank: bank}
 }
 
 // PruneBalances burns all coins owned by the account.
-func (b CoinBurner) PruneBalances(ctx sdk.Context, address sdk.AccAddress) error {
-	if amt := b.bank.GetAllBalances(ctx, address); !amt.IsZero() {
-		if err := b.bank.SendCoinsFromAccountToModule(ctx, address, types.ModuleName, amt); err != nil {
-			return sdkerrors.Wrap(err, "prune account balance")
-		}
-		if err := b.bank.BurnCoins(ctx, types.ModuleName, amt); err != nil {
-			return sdkerrors.Wrap(err, "burn account balance")
-		}
+func (b VestingCoinBurner) PruneBalances(ctx sdk.Context, existingAcc authtypes.AccountI) error {
+	v, ok := existingAcc.(vestingexported.VestingAccount)
+	if !ok {
+		return types.ErrAccountExists.Wrapf("refusing to overwrite special account type:: %T", existingAcc)
+	}
+
+	ctx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
+	coinsToBurn := sdk.NewCoins()
+	for _, orig := range v.GetOriginalVesting() { // focus on the coin denoms that were setup originally; getAllBalances has some issues
```

Comment:
Nice addition. This is limited to the number of coins that we set in the original vesting creation transaction, and thus must be some reasonable number.